### ISSUE TITLE
Permission \'iam.serviceAccounts.getAccessToken\' denied on resource (or it may not exist).

### ISSUE THREAD

**Issue by durgesh-sahani** [google-github-actions/ssh-compute#60]
### TL;DR

google-github-actions/ssh-compute failed with: failed to execute gcloud command `gcloud compute ssh *** --zone *** --ssh-key-file /tmp/4de5e605663603be1a4b13fa/google_compute_engine --quiet --tunnel-through-iap --project *** --command echo Hello world`: ERROR: (gcloud.compute.ssh) There was a problem refreshing your current auth tokens: Unable to acquire impersonated credentials

{
  "error": {
    "code": 403,
    "message": "Permission \'iam.serviceAccounts.getAccessToken\' denied on resource (or it may not exist).",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "IAM_PERMISSION_DENIED",
        "domain": "iam.googleapis.com",
        "metadata": {
          "permission": "iam.serviceAccounts.getAccessToken"
        }
      }
    ]
  }
}

### Action YAML

```yaml
name: Deploy PHP to Compute Engine

on:
  push:
    branches:
      - test
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

jobs:
  deploy:
    # Add 'id-token' with the intended permissions for workload identity federation
    permissions:
      contents: 'read'
      id-token: 'write'
      
    runs-on: ubuntu-latest

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - id: 'auth'
        name: 'Authenticate to Google Cloud'
        uses: 'google-github-actions/auth@v1'
        with:
          workload_identity_provider: '${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}'
          service_account: '${{ secrets.GCP_SERVICE_ACCOUNT }}'

      - name: Set up PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: '7.4'

      - name: Install dependencies
        run: composer install

      - id: 'compute-ssh'
        name: Deploy to Compute Engine
        uses: 'google-github-actions/ssh-compute@v0'
        with:
          project_id: ${{ secrets.GCP_PROJECT_ID }}
          zone: ${{ secrets.GCP_ZONE }}
          ssh_private_key: ${{ secrets.GCP_SSH_PRIVATE_KEY }}
          instance_name: ${{ secrets.GCP_INSTANCE_NAME }}
          command: 'echo Hello world'
```

**Comment by sethvargo** [comment]
Hi @durgesh-sahani - this error is most likely because of misconfigured Workload Identity Federation. Please follow the [Troubleshooting guide](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md). If that does not resolve the issue, please provide the [debug logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging).

**Comment by durgesh-sahani** [comment]
Hello @sethvargo, I think Workload Identity Federation worked and authenticated successfully but wile doing SSH it failed please check the attached log here:
![image](https://github.com/google-github-actions/ssh-compute/assets/38831854/3b0ad37d-7f72-4b2e-98a4-67595127938f)

**Comment by sethvargo** [comment]
Hi @durgesh-sahani - as noted in the [Troubleshooting guide](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md), the auth action for WIF only configures the authentication; it does not (and cannot) validate if the provided WIF and service account information is valid. To force a verification, follow the instructions in the Troubleshooting guide to ask the `auth` action to print out an access token or an identity token.

**Comment by durgesh-sahani** [comment]
Hello @sethvargo you are right, I have added `token_format: 'access_token'`. Now I'm getting the same error.

`
[deploy](https://github.com/MyAtOnce/git-action-php-test/actions/runs/6720384875/job/18263819927#step:5:46)
google-github-actions/auth failed with: retry function failed after 4 attempts: failed to generate Google Cloud access token for ***: (403) {
  "error": {
    "code": 403,
    "message": "Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "IAM_PERMISSION_DENIED",
        "domain": "iam.googleapis.com",
        "metadata": {
          "permission": "iam.serviceAccounts.getAccessToken"
        }
      }
    ]
  }
}
`

I have added following role to the service account 
- role: roles/iam.serviceAccountAdmin, role: roles/iam.serviceAccountAdmin, role: roles/iam.workloadIdentityUser

Still not sure why I'm getting this error.
